### PR TITLE
Fix BLE testcase setup and teardown

### PR DESCRIPTION
**Pull request by danhorner**
Fix BLE testcase setup and teardown

Description
-----------
Miscellaneous fixes for testcases encountered during testing with ESP-IDF FreeRTOS. Check return values, initialize BLE as needed, and use some older APIs during assertions.

Checklist:
----------

- [ X] I have tested my changes. No regression in existing tests.
- [ X] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/platform/test/iot_test_platform_threads.c [hunk 1]
```diff
@@ -119,23 +119,7 @@ TEST( UTIL_Platform_Threads, IotThreads_CreateDetachedThread )
...
-            &xTaskDetails,
-
-            /* Include the stack high water mark value in the
-             * TaskStatus_t structure. */
-            pdTRUE,
-            /* Include the task state in the TaskStatus_t structure. */
-            eInvalid );
-
-        *( int32_t * ) param = xTaskDetails.uxCurrentPriority;
+        *( int32_t * ) param = uxTaskPriorityGet( NULL );
```
Comment (ravibhagavandas) [inline libraries/abstractions/platform/test/iot_test_platform_threads.c]: Will this be always enabled ? Checking freertos documentation: https://www.freertos.org/a00128.html INCLUDE_uxTaskPriorityGet  should be set to 1.
Previous code had a guard for configUSE_TRACE_FACILITY which was required to be enabled for `vTaskGetInfo()`

File: libraries/abstractions/platform/test/iot_test_platform_threads.c [hunk 2]
```diff
@@ -186,23 +170,7 @@ TEST( UTIL_Platform_Threads, IotThreads_CreateDetachedThread )
...
-            &xTaskDetails,
-
-            /* Include the stack high water mark value in the
-             * TaskStatus_t structure. */
-            pdTRUE,
-            /* Include the task state in the TaskStatus_t structure. */
-            eInvalid );
-
-        *( int32_t * ) param = 300 + xTaskDetails.usStackHighWaterMark;
+        *( int32_t * ) param = 300 + uxTaskGetStackHighWaterMark( NULL );
```
Comment (ravibhagavandas) [inline libraries/abstractions/platform/test/iot_test_platform_threads.c]: same as comment  `uxTaskPriorityGet()`

File: libraries/abstractions/platform/test/iot_test_platform_threads.c (hunk 1 above)
Comment (danhorner) [reply libraries/abstractions/platform/test/iot_test_platform_threads.c]: Thanks. Good catch. It's enabled in all of our test configurations, but I've gone ahead and updated the guard.

File: libraries/abstractions/platform/test/iot_test_platform_threads.c (hunk 2 above)
Comment (danhorner) [reply libraries/abstractions/platform/test/iot_test_platform_threads.c]: I went ahead and removed this function. It's unused.

**Comment by danhorner** [comment]
Rebase to master and merge.

**Pull request merged**